example-app/app/core/services/google-books.ts

```diff
@@ -1,22 +1,22 @@
 import 'rxjs/add/operator/map';
 import { Injectable } from '@angular/core';
-import { Http } from '@angular/http';
+import { HttpClient } from '@angular/common/http';
 import { Observable } from 'rxjs/Observable';
 import { Book } from '../../books/models/book';
 
 @Injectable()
 export class GoogleBooksService {
   private API_PATH = 'https://www.googleapis.com/books/v1/volumes';
 
-  constructor(private http: Http) {}
+  constructor(private http: HttpClient) {}
 
   searchBooks(queryTitle: string): Observable<Book[]> {
     return this.http
-      .get(`${this.API_PATH}?q=${queryTitle}`)
-      .map(res => res.json().items || []);
+      .get<Book[]>(`${this.API_PATH}?q=${queryTitle}`)
```

Comment:
The response is an object that contains items, which is an array of books. Pass that as the type

```
.get<{ items: Book[] }>
```

Then you won't have to type `books` as `any`, because `items` will be inferred as `Book[]` and just use `res => res.items || []`